include/nbl/video/ISwapchain.h

```diff
@@ -85,7 +93,14 @@ class NBL_API ISwapchain : public core::IReferenceCounted, public IBackendObject
         virtual ~ISwapchain() = default;
 
         SCreationParams m_params;
-        images_array_t m_images;
+        uint32_t m_imageCount;
+        std::atomic_uint32_t m_imageExists = 0;
+
+        void freeImageExists(uint32_t ix) { m_imageExists.fetch_and(~(1U << ix)); }
+
+        // Returns false if the image already existed
+        bool setImageExists(uint32_t ix) { return (m_imageExists.fetch_or(1U << ix) & (1U << ix)) == 0; }
+
```

Comment:
great now do
```cpp
static inline constexpr uint32_t MaxImages = sizeof(m_imageExists)*8u;
```

then sprinkle some validation of parameters

also make swapchain image count a `uint8_t` in the creation params (unless I already did in #395)